### ISSUE TITLE
Misleading job status - running

### ISSUE THREAD

**Issue by sheerun** [hashicorp/nomad#1085]
### Nomad version
`v0.3.2-dev`

### Issue

I run following job:

```
{
  "Job": {
    "ID": "1234",
    "Name": "1234",
    "Region": "global",
    "Datacenters": ["dc1"],
    "Priority": 50,
    "Type": "batch",
    "TaskGroups": [
      {
        "Count": 2,
        "Name": "default",
        "Tasks": [
          {
            "Name": "default",
            "Driver": "raw_exec",
            "Config": {
              "Command": "/usr/bin/env",
              "Args": ["bash", "-c", "date && sleep 5s"]
            },
            "Resources": {
              "CPU": 100,
              "MemoryMB": 10,
              "DiskMB": 10
            },
            "LogConfig": {
              "MaxFileSizeMB": 1,
              "MaxFiles": 1
            }
          }
        ]
      }
    ]
  }
}
```

After all allocations are completed, I query the job status and get:

```
[
  {
    "ID": "1234",
    "ParentID": "",
    "Name": "1234",
    "Type": "batch",
    "Priority": 50,
    "Status": "running",
    "StatusDescription": "",
    "CreateIndex": 347,
    "ModifyIndex": 349
  }
]
```

The job is in `running` status that suggests some of the allocations haven't finished.

**Comment by dadgar** [comment]
Hmm could you try on a newer version. Just tried to reproduce it and I could not:
```
curl -XPUT -d @repro.json http://127.0.0.1:4646/v1/job/periodic
Job ID does not matchvagrant@nomad:/opt/gopath/src/github.com/hashicorp/nomad$ curl -XPUT -d @repro.json http://127.0.0.1:4646/v1/job/1234
{"EvalID":"28551f78-c1dd-4e30-760a-73894bf904ef","EvalCreateIndex":7,"JobModifyIndex":6,"Index":7,"LastContact":0,"KnownLeader":false}vagrant@nomad:/opt/gopath/src/github.com/hashicorp/nomad$ nomad status 1234
ID          = 1234
Name        = 1234
Type        = batch
Priority    = 50
Datacenters = dc1
Status      = running
Periodic    = false

==> Evaluations
ID        Priority  Triggered By  Status
28551f78  50        job-register  complete

==> Allocations
ID        Eval ID   Node ID   Task Group  Desired  Status
b29049ee  28551f78  44289a9c  default     run      running
c7bfb58c  28551f78  44289a9c  default     run      running
vagrant@nomad:/opt/gopath/src/github.com/hashicorp/nomad$ nomad status 1234
ID          = 1234
Name        = 1234
Type        = batch
Priority    = 50
Datacenters = dc1
Status      = dead
Periodic    = false

==> Evaluations
ID        Priority  Triggered By  Status
28551f78  50        job-register  complete

==> Allocations
ID        Eval ID   Node ID   Task Group  Desired  Status
b29049ee  28551f78  44289a9c  default     run      complete
c7bfb58c  28551f78  44289a9c  default     run      complete
$ curl http://127.0.0.1:4646/v1/jobs
[
  {
    "ModifyIndex": 13,
    "ID": "1234",
    "ParentID": "",
    "Name": "1234",
    "Type": "batch",
    "Priority": 50,
    "Status": "dead",
    "StatusDescription": "",
    "CreateIndex": 6
  }
]
```

**Comment by sheerun** [comment]
It's because you are using old version, not me :) The issue is with `v0.3.2-dev`, you are using `0.3.1`

**Comment by sheerun** [comment]
Hmm. Actually you were using `0.3.2-dev` few commits ahead. Indeed the job changes status to `dead`.

Should it be `completed` though?

Also, the job description I sent because invalid - I needed to change `Command` to `command` and `Args` to `args`. Isn't it breaking compatibility?

**Comment by iverberk** [comment]
@sheerun the [docs](https://www.nomadproject.io/docs/drivers/docker.html) show lowercase for all keys so it's not strictly breaking compatibility. We are just enforcing stricter checking of driver configuration now. 

But I see how this might suprise you. @dadgar @diptanu I actually thought about doing case-insensitive matching but I believe the case-information is lost after mapstructure decoding, so I'm not sure if we can and/or want to do a case-insensitive comparison of keys.

**Comment by sheerun** [comment]
The docs show examples in .hcl format and the only example in JSON format shows "Command" from capital case:

https://www.nomadproject.io/docs/jobspec/json.html

The casing between .json and .hcl is really confisuing, by the way.. Even key names don't match.

**Comment by dadgar** [comment]
@sheerun Jobs can be in the following states: [pending, running, dead]. Allocations have [pending, running, complete, failed]. 

I apologize for the incorrect JSON documentation. It will be fixed when we release 0.3.2.
